### PR TITLE
Add client-side support for RP-Initiated logout

### PR DESCRIPTION
**Pull request by zecakeh**
According to the [OpenID Connect spec](https://openid.net/specs/openid-connect-rpinitiated-1_0.html).

### CODE REVIEW

File: crates/oidc-client/src/requests/rp_initiated_logout.rs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+/// be present in the query when the end user is redirected to the
+/// `post_logout_redirect_uri`.
+///
+/// # Errors
+///
+/// Returns an error if preparing the URL fails.
+///
+/// [`VerifiedClientMetadata`]: oauth2_types::registration::VerifiedClientMetadata
+/// [`ClientErrorCode`]: oauth2_types::errors::ClientErrorCode
+#[allow(clippy::too_many_lines)]
```
[the remark this answers is not in the document]
Comment (sandhose) [reply crates/oidc-client/src/requests/rp_initiated_logout.rs:91]: well tbf there are other places where we have this even though it is not needed, we might as well cleanup later instead

File: crates/oauth2-types/src/oidc.rs [hunk 2]
```diff
@@ -1057,6 +1069,66 @@ fn validate_signing_alg_values_supported<'a>(
...
+}
+
+impl fmt::Debug for RpInitiatedLogoutRequest {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("RpInitiatedLogoutRequest")
+            .field("logout_hint", &self.logout_hint)
+            .field("post_logout_redirect_uri", &self.post_logout_redirect_uri)
+            .field("ui_locales", &self.ui_locales)
+            .finish_non_exhaustive()
+    }
```
[the remark this answers is not in the document]
Comment (zecakeh) [reply crates/oauth2-types/src/oidc.rs:1129]: Tbh I copied the behavior of AuthorizationRequest, that hides the client ID and the ID token hint too. I'm not sure why we did that but I guess it's mostly to hide sensitive information.

File: crates/oidc-client/src/requests/rp_initiated_logout.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (zecakeh) [reply crates/oidc-client/src/requests/rp_initiated_logout.rs:91]: Oops right, probably some copy-paste leftover.

[the remark this answers is not in the document]
Comment (zecakeh) [reply crates/oidc-client/src/requests/rp_initiated_logout.rs]: Removed

**Comment by zecakeh** [comment]
Added tests for generating the URL. I don't know if you had anything else in mind?

**Comment by sandhose** [comment]
That's what I expected, thanks!

**Pull request merged**